src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/RoleToAccessProfilesMappingsGenerator.java

```diff
@@ -0,0 +1,126 @@
+package uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.data.caseaccess.CachedCaseRoleRepository;
+import uk.gov.hmcts.ccd.data.caseaccess.CaseRoleRepository;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.AccessProfile;
+import uk.gov.hmcts.ccd.domain.model.definition.AccessControlList;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.data.caseaccess.GlobalCaseRole.CREATOR;
+
+@Component
+public class RoleToAccessProfilesMappingsGenerator {
+
+    protected static final String IDAM_PREFIX = "idam:";
+    private static final String CASE_ROLE_ID_REGEX = "^(\\[[A-Za-z]+\\])$";
+
+    private final CaseRoleRepository caseRoleRepository;
+
+    public RoleToAccessProfilesMappingsGenerator(@Qualifier(CachedCaseRoleRepository.QUALIFIER)
+                                                     CaseRoleRepository caseRoleRepository) {
+        this.caseRoleRepository = caseRoleRepository;
+    }
+
+    public List<AccessProfile> generate(CaseTypeDefinition caseTypeDefinition) {
+
+        Set<String> caseRoles = generateCaseRoles(caseTypeDefinition);
```

Comment:
I see that this is making a remote call to Def Store. That is not good performance wise. Have you checked whether the case roles by any change are included in the `CaseTypeDefinition.accessControlLists` ? If yes better extracting them from there.
Otherwise, we need to raise a ticket to add the case roles to the case type definition. Let me know please